platform/security/platform-security-core-api/src/main/java/ddf/security/SubjectUtils.java

```diff
@@ -161,36 +163,44 @@ public static String filterDN(X500Principal principal, Predicate<RDN> predicate)
     }
 
     public static String getEmailAddress(Subject subject) {
-        if (subject == null) {
-            LOGGER.debug("Incoming subject was null, cannot look up email address.");
+        List<String> values = getAttribute(subject, EMAIL_ADDRESS_CLAIM_URI);
+
+        if (values.isEmpty()) {
             return null;
         }
 
+        return values.get(0);
+    }
+
+    public static List<String> getAttribute(Subject subject, String key) {
```

Comment:
JavaDoc comments would be nice here, especially since there's a couple of different scenarios where the return list can be empty. We should also add `@Nullable` on `subject` since it is allowed and add a `Validate.notNull()` on the `key` parameter.